### PR TITLE
Add endgame

### PR DESCRIPTION
**Pull request by mark-todd**

### CODE REVIEW

File: examples/test_output_excel.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        run_data[state.current_time] = state_data
+    return run_data
+
+run_iters = 5
+with tempfile.TemporaryFile() as f:
+    simulation.save(f)
+    data = [run_sim(Simulation.restore(f)) for _ in range(run_iters)]
+
+
+data_by_year: dict[Year, dict[AgeStart, list[Prevalence]]] = {}
```
Comment (mkcinu) [inline examples/test_output_excel.py:40]: Possibly useless here, but I've just remembered about `itertools.groupby` - have a look maybe you'll find it useful.

File: epioncho_ibm/endgame_simulation.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            assert params
+            current_params = params[-1]
+            current_treatment = current_params.params.treatment
+            assert (
+                not current_treatment or current_treatment.stop_time < time_of_change
+            ), "Overlapping treatment found!"
+
+            program = next(programs)
+
+            # TODO: what about changing other parameters? It was decided that they can't change I suppose...
```
Comment (mkcinu) [inline epioncho_ibm/endgame_simulation.py:112]: TODO's should probably disappear.

File: epioncho_ibm/state/people.py [hunk 3]
```diff
@@ -270,3 +270,32 @@ def process_deaths(self, people_to_die: Array.Person.Bool, gender_ratio: float):
             self.worms.fertile[:, people_to_die] = 0
             self.worms.infertile[:, people_to_die] = 0
         self.delay_arrays.process_deaths(people_to_die)
+
+    def get_people_for_age_group(self, age_start: float, age_end: float) -> "People":
+        rel_ages = np.logical_and(self.ages>=age_start, self.ages<age_end)
```
Comment (mkcinu) [inline epioncho_ibm/state/people.py:275]: Or: `(self.ages >= age_start) & (self.ages < age_end)`

Comment (mark-todd) [reply epioncho_ibm/state/people.py:275]: Ah I tried `and` but not `&`! Will check it out

**Pull request merged**